cmd/backup/encrypt_archive.go

```diff
@@ -4,20 +4,87 @@
 package main
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
 	"io"
 	"os"
 	"path"
 
+	"github.com/ProtonMail/go-crypto/openpgp/armor"
 	openpgp "github.com/ProtonMail/go-crypto/openpgp/v2"
 	"github.com/offen/docker-volume-backup/internal/errwrap"
 )
 
-// encryptArchive encrypts the backup file using PGP and the configured passphrase.
-// In case no passphrase is given it returns early, leaving the backup file
+func readArmoredKeys(data []byte) (openpgp.EntityList, error) {
+	block, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	return block, nil
+}
+
+func (s *script) encryptAsymmetrically(outFile *os.File) (io.WriteCloser, func() error, error) {
+	if s.c.GpgPublicKeys == "" {
+		return nil, nil, nil
+	}
+
+	entityList, err := readArmoredKeys([]byte(s.c.GpgPublicKeys))
+	if err != nil {
+		return nil, nil, errwrap.Wrap(err, fmt.Sprintf("Error parsing key: %v", err))
+	}
+
+	armoredWriter, err := armor.Encode(outFile, "PGP MESSAGE", nil)
+	if err != nil {
+		return nil, nil, errwrap.Wrap(err, "error preparing encryption")
+	}
+
+	_, name := path.Split(s.file)
+	dst, err := openpgp.Encrypt(armoredWriter, entityList, nil, nil, &openpgp.FileHints{
+		FileName: name,
+	}, nil)
+	if err != nil {
+		return nil, nil, err
```

Comment:
This error should still be wrapped like all the others.

Reply:
I decided not to wrap it here and do it in the caller func instead. So the warping is the same for both at a single point.